podman_tutorial.md

```diff
@@ -0,0 +1,153 @@
+for podman.
+
+### Installing build and runtime dependencies
+```
+# sudo dnf install -y git runc libassuan-devel golang golang-github-cpuguy83-go-md2man glibc-static \
+                                    gpgme-devel glib2-devel device-mapper-devel libseccomp-devel \
+                                    atomic-registries iptables skopeo-containers containernetworking-cni
+```
+### Building and installing podman
+```
+# git clone https://github.com/projectatomic/libpod/ ~/src/github.com/projectatomic/libpod
+# cd !$ (or cd src/github.com/projectatomic/libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+<!-- (
+## Install podman on Ubuntu
+
+The default Ubuntu cloud image size will not allow for the following exercise to be done without increasing its
+capacity.  Be sure to add at least 5GB to the image. Instructions to do this are outside the scope of this
+tutorial.
+
+```
+# sudo apt-get update
+# sudo apt-get install libdevmapper-dev libglib2.0-dev libgpgme11-dev golang libseccomp-dev \
+                        go-md2man libprotobuf-dev libprotobuf-c0-dev libseccomp-dev
+# mkdir -p ~/src/github.com/projectatomic/
+# cd ~/src/github.com/projectatomic/
+# git clone https://github.com/projectatomic/libpod/
+#
+# cd libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo mkdir -p /etc/containers
+# sudo bash -c 'cat <<EOF > /etc/containers/registries.conf
+registries.search
+registries = ['docker.io', 'registry.fedoraproject.org']
+EOF
+'
+# sudo bash -c 'cat <<EOF > /etc/containers/policy.json
+ {
+     "default": [
+         {
+             "type": "insecureAcceptAnything"
+         }
+     ],
+     "transports":
+         {
+             "docker-daemon":
+                 {
+                     "": [{"type":"insecureAcceptAnything"}]
+                 }
+         }
+ }
+ EOF
+'
+# git clone https://github.com/containernetworking/plugins.git ~/src/github.com/containernetworking/plugins
+# cd ~/src/github.com/containernetworking/plugins
+# ./build
+# sudo mkdir -p /usr/libexec/cni
+# sudo cp bin/* /usr/libexec/cni
+# git clone https://github.com/opencontainers/runc.git ~/src/github.com/opencontainers/runc
+# GOPATH=~/ make static BUILDTAGS="seccomp selinux"
+# sudo cp runc /usr/bin/runc
+# sudo mkdir -p /usr/local/libexec
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+) -->
+## Familiarizing yourself with podman
+
+### Running a sample container
+This sample container will run a very basic httpd server that serves only its index
+page.
+```
+# sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd
+```
+Note the Container ID returned by that command.
```

Comment:
This id is huge.  You might want to show an alternative example using a bash variable and explain the containerID is contained within it.  i.e.
```
ctrid=$(sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd)
```

and then in follow up commands you can drop {containerid} and replace with $ctrid. A thought for consideration.  It allows someone to just cut/paste your examples line by line rather than cut/paste first part of command, cut/paste containerid, cut/paste last part of the command.

Reply:
the problem with this approach is if podman throws an error ... and depending on the error, we may never get that shown properly.  I was contemplating an ansible playbook for automation .. thoughts on that?

Reply:
I tried playing with it, I didn't see much difference in error reporting.  I just went the cheap route and removed the last 'pd' from the command and the errors were presented the same way.    The one downside is you don't see any progress, so the command "goes dark" for quite a while.  On my machine it takes about 45 seconds on the first pull.

I really like the example, but perhaps you could add a faster/smaller one as the first one?

An ansible playbook that goes along with this tutorial would be great, but I wouldn't replace the tutorial with this.

I much prefer setting the variable, just because I'm the kind that likes to cut/paste examples directly from a source document without having to do multiple cut/pastes from different source windows from one example.  That's my $.02, I don't know what others think and your call.

Reply:
fixed per our scrum agreement